py-client/client_api.py

```diff
@@ -46,343 +61,390 @@ class AIAAClient:
+            for i in range(3):
+                vx_pad[i] = int((pad / spacing[i]) if spacing[i] > 0 else pad)
+                index_min[i] = min(max(int(point[i] - vx_pad[i]), 0), int(index_min[i]))
+                index_max[i] = max(min(int(point[i] + vx_pad[i]), int(image_size[i] - 1)), int(index_max[i]))
+
+        logger.debug('Voxel Padding: {}'.format(vx_pad))
+        logger.debug('Min Index: {}'.format(index_min))
+        logger.debug('Max Index: {}'.format(index_max))
+
+        crop_index = [0, 0, 0]
+        crop_size = [0, 0, 0]
+        crop = []
+        for i in range(3):
+            crop_index[i] = index_min[i]
+            crop_size[i] = index_max[i] - index_min[i]
+            crop.append([crop_index[i], crop_index[i] + crop_size[i]])
+        logger.debug('crop_index: {}'.format(crop_index))
+        logger.debug('crop_size: {}'.format(crop_size))
+        logger.debug('crop: {}'.format(crop))
+
+        # get bounding box
+        x1 = crop[0][0]
+        x2 = crop[0][1]
+        y1 = crop[1][0]
+        y2 = crop[1][1]
+        z1 = crop[2][0]
+        z2 = crop[2][1]
+
+        # crop
+        points[::, 0] = points[::, 0] - x1
+        points[::, 1] = points[::, 1] - y1
+        points[::, 2] = points[::, 2] - z1
+
+        cropped_image = itk_image[x1:x2, y1:y2, z1:z2]
+        cropped_size = cropped_image.GetSize()
+        logger.debug('Cropped size: {}'.format(cropped_size))
+
+        # resize
+        out_image = AIAAUtils.resample_image(cropped_image, target_size, True)
+        logger.debug('Cropped Image Size: {}'.format(out_image.GetSize()))
+        SimpleITK.WriteImage(out_image, output_file, True)
+
+        # pointsROI
+        ratio = np.divide(np.asanyarray(target_size, dtype=np.float), np.asanyarray(cropped_size, dtype=np.float))
+        points[::, 0] = points[::, 0] * ratio[0]
+        points[::, 1] = points[::, 1] * ratio[1]
+        points[::, 2] = points[::, 2] * ratio[2]
+        return points.astype(int).tolist(), crop
+
+    @staticmethod
+    def image_post_processing(input_file, output_file, crop, orig_file):
+        itk_image = SimpleITK.ReadImage(input_file)
+        orig_crop_size = [crop[0][1] - crop[0][0], crop[1][1] - crop[1][0], crop[2][1] - crop[2][0]]
+        resize_image = AIAAUtils.resample_image(itk_image, orig_crop_size, False)
+
+        orig_image = SimpleITK.ReadImage(orig_file)
+        orig_size = orig_image.GetSize()
+
+        image = SimpleITK.GetArrayFromImage(resize_image)
+        result = np.zeros(orig_size[::-1], np.uint8)
+        result[crop[2][0]:crop[2][1], crop[1][0]:crop[1][1], crop[0][0]:crop[0][1]] = image
+
+        itk_result = SimpleITK.GetImageFromArray(result)
+        itk_result.SetDirection(orig_image.GetDirection())
+        itk_result.SetSpacing(orig_image.GetSpacing())
+        itk_result.SetOrigin(orig_image.GetOrigin())
+
+        SimpleITK.WriteImage(itk_result, output_file, True)
+
+    @staticmethod
+    def get_method(server_url, selector):
```

Comment:
it is a little confusing in the first place when I see "get_method", i was thinking it will get some method implementation based on selector.

What about calling it http_get? or some other name

Reply:
done :)